### PR TITLE
Refactor to partially remove deprecated `Value` classes usage

### PR DESCRIPTION
**Pull request by Torch3333**
## Description

ScalarDB core still uses the deprecated [Value](https://github.com/scalar-labs/scalardb/blob/master/core/src/main/java/com/scalar/db/io/Value.java) interface heavily. Since I don't want to have to implement the time-related types for this deprecated interface such as `TimestampValue`, `TimeValue`, etc. I needed to replace the usage of the `Value` interface with the [Column](https://github.com/scalar-labs/scalardb/blob/master/core/src/main/java/com/scalar/db/io/Column.java) interface.
I refactored partially the integration test and some parts of the core main package.

I will refactor unit tests for the same reason in a later PR if necessary.

## Related issues and/or PRs

- https://github.com/scalar-labs/scalardb/pull/2301

## Changes made

Refactor some integration tests and some classes of the core/main package to replace the usage of the [Value](https://github.com/scalar-labs/scalardb/blob/master/core/src/main/java/com/scalar/db/io/Value.java) interface by the [Column](https://github.com/scalar-labs/scalardb/blob/master/core/src/main/java/com/scalar/db/io/Column.java) interface.

## Checklist

> The following is a best-effort checklist. If any items in this checklist are not applicable to this PR or are dependent on other, unmerged PRs, please still mark the checkboxes after you have read and understood each item.

- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation to reflect the changes.
- [x] Any remaining open issues linked to this PR are documented and up-to-date (Jira, GitHub, etc.).
- [x] Tests (unit, integration, etc.) have been added for the changes.
- [x] My changes generate no new warnings.
- [x] Any dependent changes in other PRs have been merged and published.

## Additional notes (optional)

N/A

## Release notes

N/A

### CODE REVIEW

File: integration-test/src/main/java/com/scalar/db/api/DistributedStorageColumnValueIntegrationTestBase.java [hunk 1]
```diff
@@ -143,647 +140,368 @@ public void put_WithRandomValues_ShouldPutCorrectly() throws ExecutionException
...
+        col1Value,
+        col2Value,
+        col3Value,
+        col4Value,
+        col5Value,
+        col6Value,
+        col7Value);
+  }
+
+  private void assertResult(
```
Comment (Torch3333) [inline integration-test/src/main/java/com/scalar/db/api/DistributedStorageColumnValueIntegrationTestBase.java:397]: I refactored this class's test cases assertion in a common method.

File: integration-test/src/main/java/com/scalar/db/api/DistributedStorageSinglePartitionKeyIntegrationTestBase.java [hunk 2]
```diff
@@ -154,59 +154,53 @@ public void getAndScanAndDelete_ShouldBehaveCorrectly() throws ExecutionExceptio
...
       // for scan
-      for (Value<?> partitionKeyValue : partitionKeyValues) {
+      for (Column<?> partitionKeyValue : partitionKeyValues) {
         // Arrange
         Scan scan = prepareScan(partitionKeyType, partitionKeyValue);
 
         // Act Assert
         try (Scanner scanner = storage.scan(scan)) {
-          Optional<Result> result = scanner.one();
+          Optional<Result> optResult = scanner.one();
```
Comment (komamitsu) [inline integration-test/src/main/java/com/scalar/db/api/DistributedStorageSinglePartitionKeyIntegrationTestBase.java:187]: [trivial] You used `resultOpt` at https://github.com/scalar-labs/scalardb/pull/2328/files#diff-d6e999e6390b76c30190d9b158a27634af73f3db28349b60430a5378bf681283R271. Using unified naming convention would be better? (Either of `resultOpt` or `optResult` is fine with me)

Comment (Torch3333) [reply integration-test/src/main/java/com/scalar/db/api/DistributedStorageSinglePartitionKeyIntegrationTestBase.java:187]: Thank you! I will use `optResult`.
At first, I used `resultOpt` but changed it midway to `optResult` and forgot to rename that one.

Comment (Torch3333) [reply integration-test/src/main/java/com/scalar/db/api/DistributedStorageSinglePartitionKeyIntegrationTestBase.java:187]: Revised in https://github.com/scalar-labs/scalardb/pull/2328/commits/4031a797c4ed7200c412367c1c4226c4eda1cdfe

**Review by komamitsu — APPROVED** [review]
LGTM! 👍

**Comment by Torch3333** [comment]
Eventually, I will merge this PR to the master branch and not the [add_time_related_types](https://github.com/scalar-labs/scalardb/tree/add_time_related_types) feature branch since Toshi is also working on the code this PR modifies.

I will move the PR back to draft and notify you again once it is ready for review.

**Comment by Torch3333** [comment]
@komamitsu @brfrn169 @feeblefakie 
I changed the base branch of the PR to the master branch. Please have a look.

I reverted the changes of https://github.com/scalar-labs/scalardb/pull/2328/commits/97f5c909d88bd415395aa99f4aa392352fc27c09 that should not be merged into the master branch at the moment.

**Review by brfrn169 — APPROVED** [review]
LGTM! Thank you!

**Pull request merged**